master/pkg/tasks/task_command.go

```diff
@@ -31,6 +31,7 @@ type GenericCommandSpec struct {
 // ToTaskSpec generates a TaskSpec.
 func (s GenericCommandSpec) ToTaskSpec(
 	keys *ssh.PrivateAndPublicKeys,
+	taskType model.TaskType,
```

Comment:
`keys` is only passed here because it isn't know until task spec creation time. since the other is known right off the bat, we can just specific it when the GenericCommandSpec is created, I'd think.